common/persistence/nosql/nosql_execution_store_test.go

```diff
@@ -34,108 +34,179 @@ import (
 	"github.com/uber/cadence/service/history/constants"
 )
 
-func TestCreateWorkflowExecution(t *testing.T) {
-	ctrl := gomock.NewController(t)
-	defer ctrl.Finish()
-
-	mockDB := nosqlplugin.NewMockDB(ctrl)
-	logger := log.NewNoop()
-
-	store := &nosqlExecutionStore{
-		shardID:    1,
-		nosqlStore: nosqlStore{logger: logger, db: mockDB},
-	}
-
+func TestNosqlExecutionStore(t *testing.T) {
 	ctx := context.Background()
-	request := &persistence.InternalCreateWorkflowExecutionRequest{
-		RangeID:                  123,
-		Mode:                     persistence.CreateWorkflowModeBrandNew,
-		PreviousRunID:            "previous-run-id",
-		PreviousLastWriteVersion: 456,
-		NewWorkflowSnapshot:      getNewWorkflowSnapshot(),
-	}
+	shardID := 1
 	testCases := []struct {
 		name          string
-		setupMock     func()
-		request       *persistence.InternalCreateWorkflowExecutionRequest
+		setupMock     func(*gomock.Controller) *nosqlExecutionStore
+		testFunc      func(*nosqlExecutionStore) error
 		expectedError error
 	}{
 		{
-			name: "success",
-			setupMock: func() {
+			name: "CreateWorkflowExecution success",
+			setupMock: func(ctrl *gomock.Controller) *nosqlExecutionStore {
+				mockDB := nosqlplugin.NewMockDB(ctrl)
 				mockDB.EXPECT().
 					InsertWorkflowExecutionWithTasks(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).
-					Return(nil).Times(1)
+					Return(nil)
+				return newTestNosqlExecutionStore(mockDB, log.NewNoop())
+			},
+			testFunc: func(store *nosqlExecutionStore) error {
+				_, err := store.CreateWorkflowExecution(ctx, newCreateWorkflowExecutionRequest())
+				return err
 			},
-			request:       request,
 			expectedError: nil,
 		},
 		{
-			name: "failure - workflow already exists",
-			setupMock: func() {
+			name: "CreateWorkflowExecution failure - workflow already exists",
+			setupMock: func(ctrl *gomock.Controller) *nosqlExecutionStore {
+				mockDB := nosqlplugin.NewMockDB(ctrl)
 				mockDB.EXPECT().
 					InsertWorkflowExecutionWithTasks(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).
 					Return(&persistence.WorkflowExecutionAlreadyStartedError{}).Times(1)
 				mockDB.EXPECT().IsNotFoundError(gomock.Any()).Return(false).AnyTimes() // Assuming the error is not a "not found" error
```

Comment:
Are these ever called?

Reply:
There is some common call that is invoked somewhere in the code
`func convertCommonErrors(errChecker nosqlplugin.ClientErrorChecker, operation string, err error) error {
	if errChecker.IsNotFoundError(err) {
		return &types.EntityNotExistsError{
			Message: fmt.Sprintf("%v failed. Error: %v ", operation, err),
		}
	}

	if errChecker.IsTimeoutError(err) {
		return &persistence.TimeoutError{Msg: fmt.Sprintf("%v timed out. Error: %v", operation, err)}
	}

	if errChecker.IsThrottlingError(err) {
		return &types.ServiceBusyError{
			Message: fmt.Sprintf("%v operation failed. Error: %v", operation, err),
		}
	}

	if errChecker.IsDBUnavailableError(err) {
		return &persistence.DBUnavailableError{
			Msg: fmt.Sprintf("%v operation failed. Error: %v", operation, err),
		}
	}`
 that makes mocking these calls mandatory.
 
 I have tried moving these to pure packages but figured  I would do this in the next PR with extra cases for getworkflow.